sdk/go/common/resource/plugin/langruntime_plugin.go

```diff
@@ -55,35 +56,125 @@ type langhost struct {
 // plugin could not be found, or an error occurs while creating the child process, an error is returned.
 func NewLanguageRuntime(host Host, ctx *Context, runtime, workingDirectory string, info ProgramInfo,
 ) (LanguageRuntime, error) {
-	path, err := workspace.GetPluginPath(ctx.Diag,
-		apitype.LanguagePlugin, strings.ReplaceAll(runtime, tokens.QNameDelimiter, "_"), nil, host.GetProjectPlugins())
+	attachPort, err := GetLanguageAttachPort(runtime)
 	if err != nil {
 		return nil, err
 	}
 
-	contract.Assertf(path != "", "unexpected empty path for language plugin %s", runtime)
+	var plug *plugin
+	var client pulumirpc.LanguageRuntimeClient
+	if attachPort != nil {
+		port := *attachPort
```

Comment:
Is there a reason `GetLanguageAttachPort` returns a pointer to an int?

Reply:
only so it can return "nil" but an invalid port will do so I'll change it to -1